torch/utils/data/datasets/common.py

```diff
@@ -35,3 +35,17 @@ def onerror(err : OSError):
                 yield os.path.join(path, f)
         if not recursive:
             break
+
+
+def get_file_binaries_from_pathnames(pathnames : Iterable):
+
+    if not isinstance(pathnames, Iterable):
+        warnings.warn("get_file_binaries_from_pathnames needs the input be an Iterable")
+        raise TypeError
+
+    for pathname in pathnames:
+        if not isinstance(pathname, str):
+            warnings.warn("file pathname must be string type, but got {}".format(type(pathname)))
+            raise TypeError
+
+        yield (pathname, open(pathname, 'rb'))
```

Comment:
Can you please verify that we are releasing file handler with created stream objects gets garbage collected

Reply:
Ah, good catch. I don't think all the platform will do this releasing automatically. I will update the consumer dataset (tardataset, decoder dataset, etc) to try explicitly release it if possible.

Reply:
Alternatively wrap (or find wrapper) which will take care of closing stream on object __del__

Reply:
Double checked the impl of BufferedIOBased class (which is used as streaming handle obj), it inherited from IOBased which has __del__ implemented calling close() already, we don't need to take care of this manually.